### PR TITLE
Disallow configuring a custom root CA with a custom RoundTripper

### PR DESCRIPTION
**Pull request by yec-akamai**
## 📝 Description

Disallow that presenting custom root CA and custom http transport together. Print a warning and return when this happens.

## ✔️ How to Test

```
make testunit
```

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -743,6 +744,11 @@ func NewClient(hc *http.Client) (client Client) {
 			log.Fatalf("[ERROR] Error when reading cert at %s: %s\n", certPath, err.Error())
 		}
 
+		if hc.Transport != http.DefaultTransport.(*http.Transport) {
+			log.Println("[WARN] Custom transport is not allowed with a custom root CA.")
+			return
```
[the remark this answers is not in the document]
Comment (yec-akamai) [reply client.go:749]: > so the Client can still finish initializing if this condition is met

Good point! Does it mean we won't return the function in this case, and just skip configure the root CA?

[the remark this answers is not in the document]
Comment (lgarber-akamai) [reply client.go:749]: Yup!

**Review by lgarber-akamai — APPROVED** [review]
Looks great, thanks for addressing my feedback!

**Review by ykim-akamai — APPROVED** [review]
LGTM

**Pull request merged**